Docs: Update pipeline docs (#7869)
PostHog/posthog.com: contents/docs/cdp/common-questions.md

---

> title: Troubleshooting and FAQs

Reviewer: could we add the common question around 
- missing events - @MarconLP  had that doc somewhere - how to file bug reports for those. 

Marcus might also have more common questions and FAQs

---

## My app isn't working, what do I do?

> 1. Checking that the app is enabled with the correct configuration options in the [data pipeline tab](https://us.posthog.com/apps).

Author: Added link to docs.

2. Click "Logs & metrics" and go to the `processEvent metrics` tab to check that the app is processing events without errors.
3. Check the `Logs` tab to see if there are any errors.

> 4. Go to the data management tab to check if there any [ingestion warnings](https://us.posthog.com/data-management/ingestion-warnings).

Reviewer 2: can we hyperlink to these tabs?

Author: They are separate for all of the apps sadly.

> 5. If the app relates to an external service, check that the external service is working correctly. Make a request to the relevant API endpoint or use a tool like [webhook.site](https://webhook.site/).

Reviewer: you'd use webhook.site as the destination from PostHog to check that PostHog side sends the events as expected

## How do I capture data from another application?

> We depreciated the functionality of PostHog apps that enabled you to capture data from other tools. Functionally, these apps were a cron which ran every minute (or more) and pulled data from another tool.

Reviewer 2: ```suggestion
We deprecated the functionality of PostHog apps that enable you to capture data from other tools. Functionally, these apps were a cron which ran every minute (or more) and pulled data from another tool.
```

> There are two options for recreating these functionality:

Reviewer 2: ```suggestion
There are two options for recreating this functionality:
```

1. **Set up a cron of your own**. You can use a tool like [val.town](https://val.town/) to easily set up scheduled tasks that capture data from other tools and send it to PostHog. Our tutorials on [capturing new RSS items](/tutorials/rss-item-capture) and [events from Calendly webhooks](/tutorials/calendly-webhooks) are examples of this.

2. **Use our data warehouse**. PostHog's data warehouse enables you to import and use data from apps like Stripe, Hubspot, and Postgres natively in PostHog. See our [docs](/docs/data-warehouse) for more information.

> ## How do track when users do an event for the first time?

Reviewer 2: ```suggestion
## How do I track when users do an event for the first time?
```

There are a few ways to capture a user doing an event for the first time. We cover a few of these in our tutorials on [tracking new and returning users](/tutorials/track-new-returning-users) and [first touch attribution](/tutorials/first-last-touch-attribution).

> - Create a cohort matching users who have done an event for the first time recently.

Reviewer 2: Hyperlink "cohort"
```suggestion
- Create a cohort matching users who have done an event for the first time recently.
```

> - Use a custom event to set a person or event property. For example, you can set a `first_seen` property with the `$set_once` option.

Reviewer 2: hyperlink person + event propertys

- Use [HogQL](/docs/product-analytics/sql) to query for the `min(timestamp)` of an event.

> ## How do I migrate 20M+ events or rows into PostHog?
> 
> If you're migrating a large amount of data (20 million rows or 10k requests per minute), please contact [sales@posthog.com](mailto:sales@posthog.com) to make sure you aren't rate limited.
> 
> See [our migration docs](/docs/migrate/ingest-historic-data) for more information.

Reviewer: cc @bretthoerner  - we tell them to add the historical flag already, correct? Let's document this here or in the linked doc

```suggestion
## How do I migrate events or rows into PostHog?

TODO: set the historical flag on your batches

If you're migrating a large amount of data (20 million rows or 10k requests per minute), please contact [sales@posthog.com](mailto:sales@posthog.com) to make sure you aren't rate limited.

See [our migration docs](/docs/migrate/ingest-historic-data) for more information.
```

Reviewer: https://github.com/PostHog/posthog.com/pull/7917 is the flag everyone should use

## How do I do real time exports?

We don't currently support real time exports. 

If you need more than the hourly interval [batch exports](/docs/cdp/batch-exports) and are on the enterprise plan, please contact our team (or email [sales@posthog.com](mailto:sales@posthog.com)). 

If you're not on the enterprise plan, check out our [webhook issue](https://github.com/PostHog/posthog/issues/16976) on GitHub.

## Self-host app troubleshooting

If you're self-hosting, you can install a data connection by pasting a link to its public repository, or write your own app directly in PostHog.

> If you're having issues getting apps to work on your self-hosted instance of PostHog, check out our [troubleshooting guide](/docs/cdp/enabling).

Reviewer: could we advise people don't use this. Whenever we change the apps functionality they need to update their apps and e.g. we'll be taking away runEveryX apps code soon regardless if self-hosted users have code for that or not. Instead like for everyone else I suggest they filter and modify the events before sending to PostHog and run cron jobs out side of their PostHog instance - it's also better for them as it can cause their instance to have a bad time
